### PR TITLE
feat(progress-indicator): Add common progress indicator interface

### PR DESCRIPTION
**Pull request by allan-chen**
Project import generated by Copybara.

### CODE REVIEW

File: packages/material-components-web/index.ts [hunk 1]
```diff
@@ -38,6 +38,7 @@ import * as list from '@material/list/index';
 import * as menuSurface from '@material/menu-surface/index';
 import * as menu from '@material/menu/index';
 import * as notchedOutline from '@material/notched-outline/index';
+import * as progressIndicator from '@material/progress-indicator/index';
```
Comment (patrickrodee) [inline packages/material-components-web/index.ts]: I don't think this is necessary. We only use this internally.

File: scripts/webpack/js-bundle-factory.js [hunk 2]
```diff
@@ -184,6 +184,7 @@ class JsBundleFactory {
         menu: getAbsolutePath('/packages/mdc-menu/index.ts'),
         menuSurface: getAbsolutePath('/packages/mdc-menu-surface/index.ts'),
         notchedOutline: getAbsolutePath('/packages/mdc-notched-outline/index.ts'),
+        progressIndicator: getAbsolutePath('/packages/mdc-progress-indicator/index.ts'),
```
Comment (patrickrodee) [inline scripts/webpack/js-bundle-factory.js]: I also don't think this is necessary.

File: packages/material-components-web/index.ts (hunk 1 above)
Comment (allan-chen) [reply packages/material-components-web/index.ts]: Should we follow convention of exporting every package, even ones that may not be used directly (i.e. base). Leaving it out would require adding some whitelists to the dependency test just for this package

File: scripts/webpack/js-bundle-factory.js (hunk 2 above)
Comment (allan-chen) [reply scripts/webpack/js-bundle-factory.js]: I agree - webpack doesn't need this

File: packages/material-components-web/index.ts (hunk 1 above)
Comment (patrickrodee) [reply packages/material-components-web/index.ts]: Updating the dependency test would be good. I'd prefer that over unnecessarily exporting the progress indicator interface.

File: scripts/check-pkg-for-release.js [hunk 3]
```diff
@@ -126,14 +133,17 @@ function checkDependencyAddedInWebpackConfig() {
...
-  const jsconfig = WEBPACK_CONFIG.find((value) => {
-    return value.name === 'main-js-a-la-carte';
-  });
-  const nameCamel = camelCase(CLI_PACKAGE_JSON.name.replace('@material/', ''));
-  assert.notEqual(typeof jsconfig.entry[nameCamel], 'undefined',
-    'FAILURE: Component ' + CLI_PACKAGE_JSON.name + ' javascript dependency is not added to webpack ' +
-    'configuration. Please add ' + nameCamel + ' to ' + WEBPACK_CONFIG_RELATIVE_PATH + '\'s js-components ' +
-    'entry before commit.');
+  const name = getPkgName();
+  if (JS_WHITELIST.indexOf(name) === -1) {
```
Comment (patrickrodee) [inline scripts/check-pkg-for-release.js]: Let's restructure this to early exit via `return` if the name is included in the whitelist.

File: scripts/check-pkg-for-release.js [hunk 4]
```diff
@@ -162,10 +172,13 @@ function checkDependencyAddedInMDCPackage() {
 }
 
 function checkPkgDependencyAddedInMDCPackage() {
-  assert.notEqual(typeof MASTER_PACKAGE_JSON.dependencies[CLI_PACKAGE_JSON.name], 'undefined',
-    'FAILURE: Component ' + CLI_PACKAGE_JSON.name + ' is not a dependency for MDC Web. ' +
-    'Please add ' + CLI_PACKAGE_JSON.name +' to ' + MASTER_PACKAGE_JSON_RELATIVE_PATH +
-    '\' dependencies before commit.');
+  const name = getPkgName();
+  if (!CSS_WHITELIST.filter((x) => JS_WHITELIST.includes(x)).includes(name)) {
```
Comment (patrickrodee) [inline scripts/check-pkg-for-release.js]: Same comment as above: favor early exits

File: packages/material-components-web/index.ts (hunk 1 above)
Comment (allan-chen) [reply packages/material-components-web/index.ts]: Done - let's continue further discussions in #5564

File: scripts/check-pkg-for-release.js (hunk 4 above)
Comment (allan-chen) [reply scripts/check-pkg-for-release.js]: Just saw this - will make the changes in the new PR

File: scripts/check-pkg-for-release.js (hunk 3 above)
Comment (allan-chen) [reply scripts/check-pkg-for-release.js]: Done

**Comment by allan-chen** [comment]
Continuing in #5564

**Pull request closed without merging**